### PR TITLE
fix/TEC-3232-scheduled-imports

### PR DESCRIPTION
**Pull request by mitogh**
Make sure all pending jobs are selected when processing the EA Cron

Task: [TEC-3232]

[TEC-3232]: https://moderntribe.atlassian.net/browse/TEC-3232

### CODE REVIEW

**Comment by lucatume** [comment]
@mitogh this PR is based on `master`, is this the correct branch?

**Comment by mitogh** [comment]
For now yes we are waiting on a release branch or if are holding the PR, it was based from an old branch as is the one that Loxi is using at the moment.

File: src/Tribe/Aggregator/Cron.php [hunk 1]
```diff
@@ -401,31 +399,34 @@ public function verify_fetching_from_service() {
...
-				'explicit-no-batch-push-support' => array(
-					'key' => '_tribe_aggregator_allow_batch_push',
-					'value' => '1',
-					'compare' => '!=',
-				),
-			),
+				],
+				[
+					'relation'                        => 'OR',
+					// if not specified then assume batch push is not supported
```
Comment (lucatume) [inline src/Tribe/Aggregator/Cron.php]: All comments, anywhere, should stick w/ WordPress coding standards: First word cap, full-stop at the end.
```suggestion
					// If not specified then assume batch push is not supported.
```

File: src/Tribe/Aggregator/Cron.php [hunk 2]
```diff
@@ -401,31 +399,34 @@ public function verify_fetching_from_service() {
...
+					'no-batch-push-support-specified' => [
+						'key'     => '_tribe_aggregator_allow_batch_push',
+						'value'   => 'bug #23268',
+						'compare' => 'NOT EXISTS',
+					],
+					// if specified and not `1` then batch push is not supported
```
Comment (lucatume) [inline src/Tribe/Aggregator/Cron.php]:
```suggestion
					// If specified, and not `1`, then batch push is not supported.
```

File: src/Tribe/Aggregator/Record/Queue.php [hunk 3]
```diff
@@ -182,12 +181,16 @@ public function count() {
 	}
 
 	/**
-	 * Shortcut to check if this queue is empty.
+	 * Shortcut to check if this queue is empty or it has a null process
```
Comment (lucatume) [inline src/Tribe/Aggregator/Record/Queue.php]:
```suggestion
	 * Shortcut to check if this queue is empty or it has a null process.
```

File: src/Tribe/Aggregator/Record/Queue_Cleaner.php [hunk 4]
```diff
@@ -6,10 +6,10 @@ class Tribe__Events__Aggregator__Record__Queue_Cleaner {
 	/**
 	 * Default is 12hrs.
 	 *
-	 * @var int The time a record is allowed to stall before havint its status set to to failed since its creation in
+	 * @var int The time a record is allowed to stall before have the status set to to failed since its creation in
 	 *          seconds.
 	 */
-	protected $time_to_live = 43200; // For pre-PHP 5.6 compat, we do not define as 12 * HOUR_IN_SECONDS
+	protected $time_to_live = HOUR_IN_SECONDS * 12; // For pre-PHP 5.6 compat, we do not define as 12 * HOUR_IN_SECONDS
```
Comment (lucatume) [inline src/Tribe/Aggregator/Record/Queue_Cleaner.php]: No need to keep pre-5.6 compat anymore.
```suggestion
	protected $time_to_live = HOUR_IN_SECONDS * 12;
```

**Comment by lucatume** [comment]
@mitogh: I've re-started the tests as it seems a REST API issue not related to the changes.
Please make sure tests pass before merging.

**Pull request merged**